receiver/swohostmetricsreceiver/internal/providers/installedsoftware/provider_windows.go

```diff
@@ -87,20 +89,18 @@ func (provider *windowsProvider) GetSoftware() ([]InstalledSoftware, error) {
 	return values, nil
 }
 
-func processRegistryReader(registryReader registry.Reader) ([]InstalledSoftware, error) {
+func processRegistryReader(registryReader registry.Reader, logger *zap.Logger) ([]InstalledSoftware, error) {
 	values := []InstalledSoftware{}
 	uninstallKeys, err := registryReader.GetSubKeys()
 	if err != nil {
-		message := "Windows installed software can not be obtained, failed to get installed software registry keys"
-		zap.L().Error(message, zap.Error(err))
-		return values, fmt.Errorf("%s:%w", message, err)
+		return values, fmt.Errorf("Windows installed software can not be obtained: %w", err)
 	}
 
 	for _, swKeyName := range uninstallKeys {
 		registryValues, err := registryReader.GetKeyValues(swKeyName, []string{"DisplayName", "DisplayVersion", "Publisher", "InstallDate"})
 		if err != nil {
-			message := fmt.Sprintf("Unable to read from the %s registry key", swKeyName)
-			zap.L().Warn(message, zap.Error(err))
+			message := fmt.Sprintf("unable to read from the %s registry key", swKeyName)
+			logger.Warn(message, zap.Error(err))
```

Comment:
```suggestion
			logger.Warn("unable to read from registry", zap.String("registry_key", swKeyName), zap.Error(err))
```
Could something like this work?